src/App.tsx

```diff
@@ -1,25 +1,69 @@
-import React, { useState } from 'react';
+import React, { useCallback, useEffect, useState } from 'react';
 import './App.scss';
 import './styles/general.scss';
+import 'bulma';
 import { TodoList } from './components/TodoList';
 import { CurrentUser } from './components/CurrentUser';
+import { getTodos } from './api';
 
 const App: React.FC = () => {
   const [
     selectedUserId,
-    // setSelectedUserId,
+    setSelectedUserId,
   ] = useState(0);
 
+  const [todos, setTodos] = useState<Todo[]>([]);
+  const [filteredTodos, setFilteredTodos] = useState<Todo[]>([]);
+
+  const filterTodos = useCallback((source: string, value: string) => {
+    setFilteredTodos(todos
+      .filter(todo => {
+        if (source === 'input') {
+          return todo.title.includes(value);
+        }
+
+        switch (value) {
+          case 'active':
+            return todo.completed === false;
+          case 'completed':
+            return todo.completed === true;
+          case 'all':
+          default:
+            return todo;
+        }
```

Comment:
```suggestion
      .filter(todo => {
        switch (value) {
          case 'active':
            return todo.title.toLowerCase().includes(value.toLowerCase()) && !todo.completed;
          case 'completed':
            return todo.title.toLowerCase().includes(value.toLowerCase()) && todo.completed;
          case 'all':
            return todo.title.toLowerCase().includes(value.toLowerCase());
          default:
            return todo;
        }
```

Reply:
Thanks. There was a problem that I've been thinking about. You just solved it.